### PR TITLE
docs: expand documentation for secrets

### PR DESCRIPTION
**Pull request by lorenzo-cavazzi**
Documentation for the new "User secrets"

### CODE REVIEW

File: docs/topic-guides/secrets/secrets.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+.. image:: ../../_static/images/secrets_add_new.png
+  :width: 85%
+  :align: center
+  :alt: Add a new secret
+
+Once you add a secret, you cannot visualize its value again for security
+reasons. You can still change it by clicking on the ``Replace`` button,
+or remove it by clicking on the ``Delete`` button. The name cannot be changed;
+should you need to rename a secret, please delete it and create a new one
+you the new name.
```
Comment (lokijuhy) [inline docs/topic-guides/secrets/secrets.rst:58]:
```suggestion
with the new name.
```

File: docs/topic-guides/secrets/secrets.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+Use secrets in sessions
+-----------------------
+
+If you need to include secrets in a session, you need to click on the Start
```
Comment (lokijuhy) [inline docs/topic-guides/secrets/secrets.rst:63]:
```suggestion
To use secrets in a session, you need to click on the Start
```

File: docs/topic-guides/secrets/secrets.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+dropdown menu and select ``Start with options``. Quick-start sessions do not
+support secrets.
+
+Once on the "Start Session" page, you can select the secrets you want to
```
Comment (lokijuhy) [inline docs/topic-guides/secrets/secrets.rst:67]:
```suggestion
Once on the "Start with options" page, you can select the secrets you want to
```

**Review by lokijuhy — CHANGES_REQUESTED** [review]
Looks good! Just a few minor suggestions

File: docs/topic-guides/secrets/secrets.rst [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+Values can be any non-empty string, including special characters. The length
+cannot exceed 5'000 characters. Should you need to store a longer value,
+consider splitting it into multiple secrets.
+
+.. image:: ../../_static/images/secrets_add_new.png
+  :width: 85%
+  :align: center
+  :alt: Add a new secret
+
+Once you add a secret, you cannot visualize its value again for security
```
Comment (rokroskar) [inline docs/topic-guides/secrets/secrets.rst:54]:
```suggestion
Once you add a secret, you cannot see its value again for security
```

File: docs/topic-guides/secrets/secrets.rst [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+service`` that a user would like to start a session with some secret mounted. 
+It uses its ``private key`` to get the ``secret key`` and uses this to decrypt 
+the outer layer of encryption of the secret. It then creates a Kubernetes 
+secret with the (now singly encrypted) user secret, which gets mounted in the 
+user session.
+On session start, an init container reads the mounted secrets, and uses the 
+``user key`` to undo the inner encryption. It then creates files inside the 
+session with the decrypted secret values.
+
+If your data is extra sensitive, consider putting already encrypted values into
```
Comment (rokroskar) [inline docs/topic-guides/secrets/secrets.rst:136]: Should we add here a blurb along the lines of: 

```suggestion
Note that although we take many precautions to decrypt secrets only when necessary, they are in plaintext inside a session. This means that they are visible to whoever has access to the infrastructure where the session is running. If your data is extra sensitive, consider putting already encrypted values into 
```

**Review by rokroskar — CHANGES_REQUESTED** [review]
Really nice docs, thanks for writing this up! Just a few minor comments and suggestions

File: docs/spelling_wordlist.txt [hunk 6]
```diff
@@ -296,6 +297,7 @@ txt
 typesystem
 Ubuntu
 ui
+unencrypted
```
Comment (sgaist) [inline docs/spelling_wordlist.txt:300]: I would suggest to remove it and use `decrypted`.

File: docs/topic-guides/secrets/secrets.rst [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+.. image:: ../../_static/images/secrets_page.png
+  :width: 85%
+  :align: center
+  :alt: User Secrets page
+
+Click on the ``Add New Secret`` button and fill in the ``Name`` and
+``Value`` fields.
+
+The name is a unique identifier for the secret, used for the file name in
+sessions. It cannot be empty and must follow validation rules:
```
Comment (sgaist) [inline docs/topic-guides/secrets/secrets.rst:42]:
```suggestion
sessions. It cannot be empty and must follow these validation rules:
```

File: docs/topic-guides/secrets/secrets.rst [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+Security Model
+--------------
+
+.. image:: ../../_static/images/secrets_encryption_decryption.gif
+  :width: 85%
+  :align: center
+  :alt: Secrets encryption scheme
+
+Renku stores secrets in its database, doubly encrypted, ensuring that no part 
+accessible from the internet other than the session has access to unencrypted
```
Comment (sgaist) [inline docs/topic-guides/secrets/secrets.rst:101]:
```suggestion
accessible from the internet other than the session has access to decrypted
```

File: docs/topic-guides/secrets/secrets.rst [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+service``, meaning only it can decrypt the ``secret key``. The users secret is 
+then encrypted again using this ``secret key`` and stored in the database.
+At this point, the ``data service`` can't decrypt the user's secret anymore, as 
+it does not know the ``secret key``.
+
+To decrypt a secret, ``secret storage service`` gets a request from ``notebooks 
+service`` that a user would like to start a session with some secret mounted. 
+It uses its ``private key`` to get the ``secret key`` and uses this to decrypt 
+the outer layer of encryption of the secret. It then creates a Kubernetes 
+secret with the (now singly encrypted) user secret, which gets mounted in the
```
Comment (sgaist) [inline docs/topic-guides/secrets/secrets.rst:132]:
```suggestion
secret with the (now encrypted once) user secret, which gets mounted in the 
```

File: docs/topic-guides/secrets/secrets.rst [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+To decrypt a secret, ``secret storage service`` gets a request from ``notebooks 
+service`` that a user would like to start a session with some secret mounted. 
+It uses its ``private key`` to get the ``secret key`` and uses this to decrypt 
+the outer layer of encryption of the secret. It then creates a Kubernetes 
+secret with the (now singly encrypted) user secret, which gets mounted in the 
+user session.
+On session start, an init container reads the mounted secrets, and uses the 
+``user key`` to undo the inner encryption. It then creates files inside the 
+session with the decrypted secret values.
+
```
Comment (sgaist) [inline docs/topic-guides/secrets/secrets.rst:137]: May I nitpick a bit ?
The Kubernetes secret is in fact only mounted in the init container that will decrypt its content. The session (if we are talking about the main container) will only see the volume containing the decrypted secrets.

**Review by sgaist — CHANGES_REQUESTED** [review]
Overall it looks nice.

The encryption / decryption gif is excellent !

File: docs/topic-guides/secrets/secrets.rst (hunk 10 above)
Comment (lorenzo-cavazzi) [reply docs/topic-guides/secrets/secrets.rst:137]: I'll let @Panaetius answer this technical aspect :grin:

**Comment by lorenzo-cavazzi** [comment]
@sgaist @rokroskar @lokijuhy Thank you for reading through this  :pray: 
I incorporated all your suggestions; please feel free to comment further should you find anything else.

@lokijuhy I added changes to the release notes here c187e4aa937efd14d660fa185623729a7a86ef79 . Mind that the link won't work until we publish the release since it's a new page.

File: docs/topic-guides/secrets/secrets.rst (hunk 10 above)
Comment (Panaetius) [reply docs/topic-guides/secrets/secrets.rst:137]: I've changed the description a bit (I think this comment meant to address the previous paragraph, not this one)

**Review by lokijuhy — APPROVED** [review]
🎉

**Review by rokroskar — APPROVED** [review]

**Pull request merged**